src/game/AI/BaseAI/PetAI.cpp

```diff
@@ -392,7 +392,7 @@ void PetAI::UpdateAI(const uint32 diff)
             {
                 if (following)
                 {
-                    m_unit->GetMotionMaster()->Clear(false);
+                    m_unit->GetMotionMaster()->Clear(false);    // does this need to be removed since MoveFollow is calling Clear?
```

Comment:
This one is a tad tricky, because currently PetAI is used also by players who are MCed and creatures who are MCed and they might have a loooot of different movement going on.